chainer/optimizer_hooks/gradient_hard_clipping.py

```diff
@@ -47,4 +48,12 @@ def __call__(self, rule, param):
             return
         with chainer.using_device(param.device):
             xp = param.device.xp
-            xp.clip(grad, self.lower_bound, self.upper_bound, out=grad)
+            # TODO(kshitij12345): Fix when chainerx.clip
+            # supports kwarg `out`.
+            if xp == backend.chainerx \
+                    or isinstance(param.grad, backend.intel64.mdarray):
+                param.grad = grad.clip(self.lower_bound, self.upper_bound)
```

Comment:
This parameter should be updated in-place. Could you fix?
```suggestion
                grad[:] = grad.clip(self.lower_bound, self.upper_bound)
```

Reply:
Fixed. Thank You.